src/protocol.rs

```diff
@@ -292,52 +292,162 @@ impl SignatureShare {
+        other_ids: Vec<ParticipantIdentifier>,
+    }
+
+    impl ParticipantConfig {
+        /// Get the [`ParticipantIdentifier`] for the owner of this config.
+        pub fn id(&self) -> ParticipantIdentifier {
+            self.id
         }
-        let ids = std::iter::repeat_with(|| ParticipantIdentifier::random(rng))
-            .take(size)
-            .collect::<Vec<_>>();
 
-        Ok((0..size)
-            .map(|i| {
-                let mut other_ids = ids.clone();
-                let id = other_ids.swap_remove(i);
-                Self { id, other_ids }
+        /// Get the [`ParticipantIdentifier`] for all other participants.
+        pub fn other_ids(&self) -> &[ParticipantIdentifier] {
+            self.other_ids.as_slice()
+        }
+
+        /// Returns a list of all participant IDs, including `self`'s.
+        pub fn all_participants(&self) -> Vec<ParticipantIdentifier> {
+            let mut participant = self.other_ids().to_owned();
+            participant.push(self.id());
+            participant
+        }
+
+        pub(crate) fn into_parts(self) -> (ParticipantIdentifier, Vec<ParticipantIdentifier>) {
+            (self.id, self.other_ids)
+        }
+
+        /// Create a new [`ParticipantConfig`].
+        ///
+        /// The protocol requires at least two participants; `other_ids` cannot
+        /// be empty. All participant identifiers must be unique.
+        pub fn new(id: ParticipantIdentifier, other_ids: &[ParticipantIdentifier]) -> Result<Self> {
+            if other_ids.is_empty() {
+                error!(
+                    "Tried to create a participant config with too few participants. There must be at least one `other_id`.",
+                );
+                Err(CallerError::ParticipantConfigError)?
+            }
+
+            // Test for uniqueness
+            let unique_set =
+                HashSet::<&ParticipantIdentifier>::from_iter(std::iter::once(&id).chain(other_ids));
+            if unique_set.len() != other_ids.len() + 1 {
+                error!(
+                    "Tried to create a participant config with a non-unique set of participants"
+                );
+                Err(CallerError::ParticipantConfigError)?
+            }
+
+            Ok(Self {
+                id,
+                other_ids: other_ids.to_vec(),
             })
-            .collect())
+        }
+
+        /// Get a list of `size` consistent [`ParticipantConfig`]s.
+        ///
+        /// Each config contains a different permutation of a single overall set
+        /// of [`ParticipantIdentifier`]s.
+        ///
+        /// **⚠️ Security warning:** This method implies the existence of a
+        /// trusted third party that generates the participant IDs. This
+        /// method must not be used if your deployment does not have a
+        /// trusted party.
+        pub fn random_quorum<R: RngCore + CryptoRng>(
```

Comment:
note: this method didn't change except for the beefed-up security warning in the docs and a call to the `new` method instead of manually constructing `Self`.